src/lib/dnssd/Discovery_ImplPlatform.cpp

```diff
@@ -633,15 +631,27 @@ bool DiscoveryImplPlatform::IsInitialized()
 
 CHIP_ERROR DiscoveryImplPlatform::ResolveNodeId(const PeerId & peerId)
 {
-    ReturnErrorOnFailure(InitImpl());
-    return mResolverProxy.ResolveNodeId(peerId);
+    // Resolve requests can only be issued once DNSSD is initialized and there is
+    // no caching currently
+    VerifyOrReturnError(mState == State::kInitialized, CHIP_ERROR_INCORRECT_STATE);
+
```

Comment:
We used to error out in the `mOperationalDelegate == nullptr` case (CHIP_ERROR_INCORRECT_STATE).  Should we keep doing that?

Reply:
I was thinking no as the delegate is now a global, so callers can/should override them. In practice the operational delegate should always be the global `AddressResolve` instance and should never change.